server/src/test/java/io/spine/examples/shareaware/server/e2e/given/EntitySubscription.java

```diff
@@ -26,52 +26,25 @@
 
 package io.spine.examples.shareaware.server.e2e.given;
 
+import io.spine.base.CommandMessage;
 import io.spine.base.EntityState;
 import io.spine.client.Client;
 import io.spine.core.UserId;
-
-import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.ExecutionException;
-
-import static io.spine.util.Exceptions.illegalStateWithCauseOf;
+import io.spine.examples.shareaware.testing.server.e2e.AsyncObserver;
 
 /**
- * Subscription for the {@code EntityState} changes.
+ * Configures the {@code AsyncObserver} with how to send a command
+ * and how to observe the entity state changes using {@code Spine} Client API.
  */
-public class EntitySubscription<S extends EntityState> {
-
-    private final ObservedEntity<S> entity = new ObservedEntity<>();
+class EntitySubscription<S extends EntityState> extends AsyncObserver<S, CommandMessage> {
 
     EntitySubscription(Class<S> entityType, Client client, UserId user) {
-        client.onBehalfOf(user)
-              .subscribeTo(entityType)
-              .observe(entity::setState)
-              .post();
-    }
-
-    /**
-     * Provides the current state of the subscribed entity.
-     */
-    public S state() {
-        return entity.state();
-    }
-
-    private static final class ObservedEntity<S extends EntityState> {
-        private CompletableFuture<S> future = new CompletableFuture<>();
-
-        private void setState(S value) {
-            if(future.isDone()) {
-                future = new CompletableFuture<>();
-            }
-            future.complete(value);
-        }
-
-        private S state() {
-            try {
-                return future.get();
-            } catch (InterruptedException | ExecutionException e) {
-                throw illegalStateWithCauseOf(e);
-            }
-        }
+        super(consumer -> client.onBehalfOf(user)
```

Comment:
Is it possible to extract these lambda expressions to methods?

Reply:
It is not possible. We can access the methods before the supertype constructor call only if they are static, but we have a generic type `S`. So in our case, we cannot extract these lambdas to methods.